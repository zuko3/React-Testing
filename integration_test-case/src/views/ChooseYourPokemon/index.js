import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchRandomPokemon } from "../../actions";

class ChooseYourPokemon extends Component {
  render() {
    const { handleClick } = this.props;
    return (
      <div>
        <p>
          Pokem ipsum dolor sit amet Mime Jr Mime Jr Watchog Nidoran Professor
          Oak Pichu. Thunder Badge Magmortar anim id est laborum Lileep Luvdisc
          Jellicent Reuniclus. Leech Life Tyrogue Squirtle Fuchsia City Jolteon
          Masquerain ut enim ad minim veniam. Silver Escape Rope Cloyster
          Roserade Cradily Feebas Youngster wants to fight. Cerulean City
          Tentacool gym Purugly Granbull Zweilous Celadon Department Store.
        </p>

        <p>
          Rage Vaporeon Solosis Makuhita Bill Lileep Jigglypuff. Volcano Badge
          Glaceon Octillery Harden Chandelure Starmie Gulpin. Quis nostrud
          exercitation ullamco laboris nisi Stunky Shuppet Pachirisu Swadloon
          Mawile anim id est laborum. Cerulean City Goldeen Water Bronzong
          Weedle Harden Swanna. Charizard Gorebyss Basculin Milotic Escape Rope
          Pikachu Gyarados.
        </p>

        <p>
          Anim id est laborum Poliwrath Shaymin Mineral Badge Chimchar Haunter
          Staravia. Dragon Rage Kricketot Growlithe what kind of Pokemon are you
          Pinsir Great Ball Oshawott. Pikachu Smoochum Raikou Cleffa Riolu
          Amoonguss Potion. Growl Abomasnow Blitzle Marill Flaaffy Tangela
          Smoochum. Gotta catch 'em all Bulbasaur Water Gun Sharpedo to catch
          them is my real test Poison Maractus.
        </p>

        <p>
          Pewter City Flaaffy Pinsir Riolu Xatu Hydro Pump Cherubi. Sed do
          eiusmod tempor incididunt Gible Darumaka Cascoon Ekans Haunter Dome
          Fossil. Bulbasaur Gible Slowpoke Archen Sewaddle Bayleef Celadon
          Department Store. Pewter City Vanillite Beautifly Alakazam grumpy old
          man who needs coffee Zangoose Clefable. Scratch Cascade Badge Trubbish
          Sceptile Shiftry Scizor Whimsicott.{" "}
        </p>
        <button data-test-id="choose-pokemon" onClick={handleClick}>
          Choose!
        </button>
      </div>
    );
  }
}

export default connect(
  null,
  { handleClick: fetchRandomPokemon }
)(ChooseYourPokemon);
