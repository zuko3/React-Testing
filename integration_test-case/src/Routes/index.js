import React from "react";
import { Route, Switch } from "react-router-dom";
import ChooseYourPokemon from "../views/ChooseYourPokemon";
import SelectedPokemon from "../views/SelectedPokemon";

export default () => (
  <Switch>
    <Route component={ChooseYourPokemon} path="/choose-your-pokemon" />
    <Route component={SelectedPokemon} path="/selected" />
  </Switch>
);
