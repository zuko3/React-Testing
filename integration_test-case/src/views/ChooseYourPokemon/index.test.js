import Routes from "../../Routes";
import apiService from "../../services/api-service";
import {
  makeMountRender,
  mocker,
  reduxify,
  snapshotify,
  ticks,
  asyncFlush
} from "../../test-utils";
import ChooseYourPokemon from "./";
import history from "../../history";

describe("<ChooseYourPokemon />", function() {
  const apiMock = apiService.mock();

  beforeEach(() => {
    mocker(apiMock).fetchRandomPokemon();
  });

  afterEach(() => {
    apiMock.reset();
  });

  it("matches snapshot", function() {
    const wrapper = makeMountRender(reduxify(ChooseYourPokemon));
    expect(snapshotify(wrapper)).toMatchSnapshot();
  });

  it("allows users to select a random pokemon", function(done) {
    const wrapper = makeMountRender(reduxify(Routes));
    history.push("/choose-your-pokemon");

    wrapper.update();
    wrapper.find('[data-test-id="choose-pokemon"]').simulate("click");

    ticks([
      () => {
        wrapper.update();
        console.log("from tick part:", wrapper.debug());
        expect(window.location.href).toBe("http://localhost/selected");
        expect(snapshotify(wrapper)).toMatchSnapshot();
        done();
      }
    ]);
  });

  it("allows users to select a random pokemon", async function() {
    const wrapper = makeMountRender(reduxify(Routes));
    history.push("/choose-your-pokemon");

    wrapper.update();
    wrapper.find('[data-test-id="choose-pokemon"]').simulate("click");

    await asyncFlush();
    wrapper.update();

    console.log("From async part:", wrapper.debug());

    expect(window.location.href).toBe("http://localhost/selected");
    expect(snapshotify(wrapper)).toMatchSnapshot();
  });
});
