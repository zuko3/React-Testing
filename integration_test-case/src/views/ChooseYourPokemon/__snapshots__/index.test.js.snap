// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseYourPokemon /> allows users to select a random pokemon 1`] = `"<div><h2>Congrats!</h2><h3>You have selected raichu!</h3><img src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/26.png\\" alt=\\"raichu\\"></div>"`;

exports[`<ChooseYourPokemon /> allows users to select a random pokemon 2`] = `"<div><h2>Congrats!</h2><h3>You have selected raichu!</h3><img src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/26.png\\" alt=\\"raichu\\"></div>"`;

exports[`<ChooseYourPokemon /> matches snapshot 1`] = `"<div><p>Pokem ipsum dolor sit amet Mime Jr Mime Jr Watchog Nidoran Professor Oak Pichu. Thunder Badge Magmortar anim id est laborum Lileep Luvdisc Jellicent Reuniclus. Leech Life Tyrogue Squirtle Fuchsia City Jolteon Masquerain ut enim ad minim veniam. Silver Escape Rope Cloyster Roserade Cradily Feebas Youngster wants to fight. Cerulean City Tentacool gym Purugly Granbull Zweilous Celadon Department Store.</p><p>Rage Vaporeon Solosis Makuhita Bill Lileep Jigglypuff. Volcano Badge Glaceon Octillery Harden Chandelure Starmie Gulpin. Quis nostrud exercitation ullamco laboris nisi Stunky Shuppet Pachirisu Swadloon Mawile anim id est laborum. Cerulean City Goldeen Water Bronzong Weedle Harden Swanna. Charizard Gorebyss Basculin Milotic Escape Rope Pikachu Gyarados.</p><p>Anim id est laborum Poliwrath Shaymin Mineral Badge Chimchar Haunter Staravia. Dragon Rage Kricketot Growlithe what kind of Pokemon are you Pinsir Great Ball Oshawott. Pikachu Smoochum Raikou Cleffa Riolu Amoonguss Potion. Growl Abomasnow Blitzle Marill Flaaffy Tangela Smoochum. Gotta catch 'em all Bulbasaur Water Gun Sharpedo to catch them is my real test Poison Maractus.</p><p>Pewter City Flaaffy Pinsir Riolu Xatu Hydro Pump Cherubi. Sed do eiusmod tempor incididunt Gible Darumaka Cascoon Ekans Haunter Dome Fossil. Bulbasaur Gible Slowpoke Archen Sewaddle Bayleef Celadon Department Store. Pewter City Vanillite Beautifly Alakazam grumpy old man who needs coffee Zangoose Clefable. Scratch Cascade Badge Trubbish Sceptile Shiftry Scizor Whimsicott. </p><button data-test-id=\\"choose-pokemon\\">Choose!</button></div>"`;
